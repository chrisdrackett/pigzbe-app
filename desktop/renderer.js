import {AppRegistry} from 'react-native';
import App from './bundle.web.js';

AppRegistry.registerComponent('App', () => App);
AppRegistry.runApplication('App', {rootTag: document.getElementById('react-root')});
