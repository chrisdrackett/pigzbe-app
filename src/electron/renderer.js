import {AppRegistry} from 'react-native';
// import App from '../../App';
import App from '../../dist/bundle.web.js';

AppRegistry.registerComponent('App', () => App);
AppRegistry.runApplication('App', {rootTag: document.getElementById('react-root')});
