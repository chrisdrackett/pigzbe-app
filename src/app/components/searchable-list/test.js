import React from 'react';
import renderer from 'react-test-renderer';
import SearchableList from './';

const props = {
    selectedKey: 'GB',
    onChangeSelection: () => {},
    items: {
        BE: "Belgium",
        DK: "Denmark",
        FR: "France",
        DE: "Germany",
        IT: "Italy",
        JP: "Japan",
        ES: "Spain",
        GB: "United Kingdom",
    }
};

describe('SearchableList', () => {
    test('renders correctly', () => {
        const tree = renderer.create(<SearchableList {...props}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
