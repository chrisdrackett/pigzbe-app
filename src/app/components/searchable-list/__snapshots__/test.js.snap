// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchableList renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(255, 255, 255)",
      "borderRadius": 10,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomColor": "rgb(127, 153, 188)",
        "borderBottomWidth": 1,
        "height": 75,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(223, 230, 238)",
          "borderRadius": 4,
          "flexDirection": "row",
          "height": 36,
          "marginLeft": 10,
          "marginRight": 10,
          "paddingLeft": 10,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/app/components/searchable-list/images/search.png",
          }
        }
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Search"
        style={
          Object {
            "flex": 1,
            "fontFamily": "Poppins",
            "fontSize": 14,
            "marginLeft": 7,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View
    style={
      Object {
        "paddingLeft": 20,
        "paddingRight": 20,
      }
    }
  >
    <RCTScrollView
      data={
        Array [
          Object {
            "key": "BE",
            "label": "Belgium",
            "last": false,
          },
          Object {
            "key": "DK",
            "label": "Denmark",
            "last": false,
          },
          Object {
            "key": "FR",
            "label": "France",
            "last": false,
          },
          Object {
            "key": "DE",
            "label": "Germany",
            "last": false,
          },
          Object {
            "key": "IT",
            "label": "Italy",
            "last": false,
          },
          Object {
            "key": "JP",
            "label": "Japan",
            "last": false,
          },
          Object {
            "key": "ES",
            "label": "Spain",
            "last": false,
          },
          Object {
            "key": "GB",
            "label": "United Kingdom",
            "last": true,
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "borderBottomColor": "rgb(127, 153, 188)",
                  "borderBottomWidth": 1,
                  "height": 49,
                  "justifyContent": "center",
                },
                Object {},
              ]
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "rgb(0, 50, 120)",
                    "fontFamily": "Poppins",
                    "fontSize": 14,
                    "fontWeight": "bold",
                  }
                }
              >
                Belgium
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "borderBottomColor": "rgb(127, 153, 188)",
                  "borderBottomWidth": 1,
                  "height": 49,
                  "justifyContent": "center",
                },
                Object {},
              ]
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "rgb(0, 50, 120)",
                    "fontFamily": "Poppins",
                    "fontSize": 14,
                    "fontWeight": "bold",
                  }
                }
              >
                Denmark
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "borderBottomColor": "rgb(127, 153, 188)",
                  "borderBottomWidth": 1,
                  "height": 49,
                  "justifyContent": "center",
                },
                Object {},
              ]
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "rgb(0, 50, 120)",
                    "fontFamily": "Poppins",
                    "fontSize": 14,
                    "fontWeight": "bold",
                  }
                }
              >
                France
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "borderBottomColor": "rgb(127, 153, 188)",
                  "borderBottomWidth": 1,
                  "height": 49,
                  "justifyContent": "center",
                },
                Object {},
              ]
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "rgb(0, 50, 120)",
                    "fontFamily": "Poppins",
                    "fontSize": 14,
                    "fontWeight": "bold",
                  }
                }
              >
                Germany
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "borderBottomColor": "rgb(127, 153, 188)",
                  "borderBottomWidth": 1,
                  "height": 49,
                  "justifyContent": "center",
                },
                Object {},
              ]
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "rgb(0, 50, 120)",
                    "fontFamily": "Poppins",
                    "fontSize": 14,
                    "fontWeight": "bold",
                  }
                }
              >
                Italy
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "borderBottomColor": "rgb(127, 153, 188)",
                  "borderBottomWidth": 1,
                  "height": 49,
                  "justifyContent": "center",
                },
                Object {},
              ]
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "rgb(0, 50, 120)",
                    "fontFamily": "Poppins",
                    "fontSize": 14,
                    "fontWeight": "bold",
                  }
                }
              >
                Japan
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "borderBottomColor": "rgb(127, 153, 188)",
                  "borderBottomWidth": 1,
                  "height": 49,
                  "justifyContent": "center",
                },
                Object {},
              ]
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "rgb(0, 50, 120)",
                    "fontFamily": "Poppins",
                    "fontSize": 14,
                    "fontWeight": "bold",
                  }
                }
              >
                Spain
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "borderBottomColor": "rgb(127, 153, 188)",
                  "borderBottomWidth": 1,
                  "height": 49,
                  "justifyContent": "center",
                },
                Object {
                  "borderBottomWidth": 0,
                },
              ]
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "rgb(0, 50, 120)",
                    "fontFamily": "Poppins",
                    "fontSize": 14,
                    "fontWeight": "bold",
                  }
                }
              >
                United Kingdom
              </Text>
              <Image
                source={
                  Object {
                    "testUri": "../../../src/app/components/searchable-list/images/tick.png",
                  }
                }
                style={
                  Object {
                    "marginRight": 15,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
